spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/SimpleJobService.java

```diff
@@ -336,17 +346,13 @@ public int countJobInstances(String name) {
 
 	@Override
 	public JobExecution getJobExecution(Long jobExecutionId) throws NoSuchJobExecutionException {
-		JobExecution jobExecution = jobExecutionDao.getJobExecution(jobExecutionId);
-		if (jobExecution == null) {
-			throw new NoSuchJobExecutionException("There is no JobExecution with id=" + jobExecutionId);
-		}
-		jobExecution.setJobInstance(Objects.requireNonNull(jobInstanceDao.getJobInstance(jobExecution)));
+		JobExecution jobExecution =  this.aggregateJobQueryDao.getJobExecution(jobExecutionId, this.schemaTarget).getJobExecution();
+		jobExecution.setJobInstance(Objects.requireNonNull(this.jobInstanceDao.getJobInstance(jobExecution)));
```

Comment:
I think you want to use `this.getJobInstance` here instead of `this.jobIinstanceDao.getJobInstance` as that uses the newly added aggregate API.

Reply:
In this case we are using the JobExecutionId to find its `JobInstance`